lib/Dialect/FIRRTL/Transforms/GrandCentral.cpp

```diff
@@ -13,548 +13,1055 @@
+            removalError = true;
+            return false;
+          });
+        })
+        .Case<MemOp>([&](auto op) {
+          AnnotationSet::removeAnnotations(op, [&](Annotation annotation) {
+            if (!annotation.isClass(
+                    "sifive.enterprise.grandcentral.AugmentedGroundType"))
+              return false;
+            op.emitOpError()
+                << "is marked as an interface element, but this does "
+                   "not make sense (is there a scattering bug or do "
+                   "you have a malformed hand-crafted MLIR circuit?)";
+            removalError = true;
+            return false;
+          });
+          AnnotationSet::removePortAnnotations(
+              op, [&](unsigned i, Annotation annotation) {
+                if (!annotation.isClass(
+                        "sifive.enterprise.grandcentral.AugmentedGroundType"))
+                  return false;
+                op.emitOpError()
+                    << "has port '" << i
+                    << "' marked as an interface element, but this does "
+                       "not make sense (is there a scattering bug or do "
+                       "you have a malformed hand-crafted MLIR circuit?)";
+                removalError = true;
+                return false;
+              });
+        })
+        .Case<FModuleOp>([&](FModuleOp op) {
+          // Handle annotations on the ports.
+          AnnotationSet::removePortAnnotations(
+              op, [&](unsigned i, Annotation annotation) {
+                if (!annotation.isClass(
+                        "sifive.enterprise.grandcentral.AugmentedGroundType"))
+                  return false;
+                auto maybeID = getID(op, annotation);
+                if (!maybeID)
+                  return false;
+                leafMap.insert({maybeID.getValue(), op.getArgument(i)});
+
+                return true;
+              });
+
+          // Handle annotations on the module.
+          AnnotationSet::removeAnnotations(op, [&](Annotation annotation) {
+            if (!annotation.isClass(
+                    "sifive.enterprise.grandcentral.ViewAnnotation"))
+              return false;
+            auto tpe = annotation.getMember<StringAttr>("type"),
+                 name = annotation.getMember<StringAttr>("name");
+            auto id = annotation.getMember<IntegerAttr>("id");
+            if (!tpe) {
+              op.emitOpError() << "has a malformed "
+                                  "'sifive.enterprise.grandcentral."
+                                  "ViewAnnotation' that "
+                                  "did "
+                                  "not contain a 'type' field with a "
+                                  "'StringAttr' value";
+              goto FModuleOp_error;
+            }
+            if (!id) {
+              op.emitOpError() << "has a malformed "
+                                  "'sifive.enterprise.grandcentral."
```

Comment:
is this really what clang-format did with these strings?

Reply:
Good eye. Yes, this is what it was doing, but only due to me writing and rewriting the error messages. I did a clean format of these after putting contiguous strings on one line and it now looks much better. :+1: